roles/network_plugin/calico/tasks/check.yml

```diff
@@ -53,7 +53,7 @@
 - name: "Check if inventory match current cluster configuration"
   assert:
     that:
-      - calico_pool_conf.spec.blockSize|string == (calico_pool_blocksize | default(kube_network_node_prefix | string))
+      - calico_pool_conf.spec.blockSize|string == (calico_pool_blocksize | default(kube_network_node_prefix) | string)
```

Comment:
Shouldn't these be compared as `int` instead of `string` ?

Reply:
Well, #8120 changed the comparison format from int to string.
To answer your question, if so, the referenced PR should be reverted and mine deleted I guess....

Reply:
> To answer your question, if so, the referenced PR should be reverted and mine changed so that..
Actually, I take back what I posted in my last comment, if so, the referenced PR should be reverted and mine deleted.

I took a look at #8119. I think the issue is a false positive as ansible key/val parser there does not do typing and passing `-e kube_network_node_prefix=26` will make `kube_network_node_prefix` be a string whereas it is expected t be an int (in inventory)

Reply:
@cristicalin What do you think ?

Reply:
You can always do the right thing :tm: in this PR and fix both the order and they type cast.
I have to admit that #8120 dropped off my radar after the premature merge.

PS: everything you pass to ansible via `-e var=value` ends up as a string, to pass different types like `int` or `bool` you have to use the json syntax like `-e '{bool_var: True, my_int_var: 23}'.

Reply:
This PR fixes the type cast (every possible variable involved is cast as string for comparison).
As mentioned, either the type cast should be kept (and my PR merged to have things fixed when `calico_pool_blocksize` is defined - as int - in inventory) or the #8120 PR should be reverted.
IMO, the best thing should be to revert the PR as there is no need to do type cast as far as the variables passed have the right/expected type.
So I'll close this PR and open another one reverting PR#8120, is it fine by you @cristicalin ?

Reply:
Forcing the typecast isn't necessarily a bad idea just that it should be the right type to avoid confusion. In this particular case I see your point that you should not supply these parameters from the command line to avoid errors in the type propagation but there may still be situations where that variable may be passed from some sort of playbook import where the type could be mangled and it is better to ensure the typecast at the place of comparison.

Alternatively we could add a validation role for input that does type checking based on something like [json-schema](https://json-schema.org/) but that is a lot more work to cover all of our public API.

Reply:
@cristicalin 
My point wasn't not to pass the parameter but to pass it the right way:
`-e '{"kube_network_node_prefix": 26}'` instead of `-e kube_network_node_prefix=26`

Reply:
There are valid situations where you cannot control the type ansible assigns to this variable, especially if you embed kubespray as part of a larger project and call `cluster.yml` with `include_playbook` and your variables are calculated or extracted from some other data source.

It is a good practice to ensure types a properly set when variables are compared to avoid issues propagating silently. As was the case with the original PR #8120 this should be fixed by setting the proper types when comparing the variables since at least `kube_network_node_prefix` is user supplied input and all user input should be sanitized.

Reply:
Fine by me. So my PR stands, it defaults `calico_pool_blocksize` to `kube_network_node_prefix` and then applies type cast.
LGTY?

Reply:
As said above, I think the typecast should be to `int` instead of `string` but the change in logic is correct.

Reply:
Sorry I misunderstood your point enforcing int type. That should fixed now